frontend/src/components/classify-failures.js

```diff
@@ -80,184 +151,173 @@ export class ClassifyFailuresTable extends React.Component {
     // maybe process values array to string format here instead of expecting caller to do it?
     let operator = (value.includes(';')) ? 'in' : 'eq';
-    this.updateFilters(filterId, field, operator, value, this.refreshResults);
-  };
+    updateFilters(filterId, field, operator, value);
+  }, [updateFilters]);
 
-  removeFilter = (filterId, id) => {
+  const removeFilter = (filterId, id) => {
     if ((id !== 'result') && (id !== 'run_id')) {   // Don't allow removal of error/failure filter
-      this.updateFilters(filterId, id, null, null, this.refreshResults);
+      updateFilters(filterId, id, null, null);
     }
   };
 
-  onSkipCheck = (checked) => {
-    let { filters } = this.state;
-    filters['result']['val'] = ('failed;error') + ((checked) ? ';skipped;xfailed' : '');
-    this.setState(
-      {includeSkipped: checked, filters},
-      this.refreshResults
-    );
-  };
+  const resultFilters = [
+    <MetaFilter
+      key="metafilter"
+      runId={run_id}
+      setFilter={setFilter}
+      activeFilters={appliedFilters}
+      onRemoveFilter={removeFilter}
+      hideFilters={['run_id', 'project_id']}
+      id={0}
+    />,
+  ];
 
-  getSelectedResults = () => {
-    const { results, rows } = this.state;
-    let selectedResults = [];
-    for (const [index, row] of rows.entries()) {
-      if (row.selected && row.parent === null) {  // rows with a parent attr are the child rows
-        selectedResults.push(results[index / 2]);  // divide by 2 to convert row index to result index
-      }
-    }
-    this.setState({selectedResults});
-  };
-
-  getResultsForTable () {
-    const filters = this.state.filters;
-    this.setState({rows: [getSpinnerRow(5)], isEmpty: false, isError: false});
-    // get only failed results
-    let params = buildParams(filters);
-    params['filter'] = toAPIFilter(filters);
-    params['pageSize'] = this.state.pageSize;
-    params['page'] = this.state.page;
-    this.setState({rows: [['Loading...', '', '', '', '']]});
-    HttpClient.get([Settings.serverUrl, 'result'], params)
-      .then(response => HttpClient.handleResponse(response))
-      .then(data => this.setState({
-        results: data.results,
-        rows: data.results.map((result, index) => resultToClassificationRow(result, index, this.setFilter)).flat(),
-        page: data.pagination.page,
-        pageSize: data.pagination.pageSize,
-        totalItems: data.pagination.totalItems,
-        totalPages: data.pagination.totalPages,
-        isEmpty: data.pagination.totalItems === 0,
-      }))
-      .catch((error) => {
-        console.error('Error fetching result data:', error);
-        this.setState({rows: [], isEmpty: false, isError: true});
-      });
-  }
+  useEffect(() => {
+    // set rows when filtered items update
+    setRows(
+      filteredResults.map(
+        (result, index) => resultToClassificationRow(result, index, setFilter)
+      )
+        .flat()
```

Comment:
instead of doing map() and then flat() - you can do flatMap()
```
setRows(
  filteredResults.flatMap((result, index) => resultToClassificationRow(result, index, setFilter))
);
```